From 329e3658e7cd06bb3ef3668e5d036a7197630730 Mon Sep 17 00:00:00 2001
From: Phoenix616 <mail@moep.tv>
Date: Mon, 20 Jan 2020 18:51:00 +0100
Subject: [PATCH] Add PlayerPortalEvent to end exit portal

---
 .../net/minecraft/server/level/EntityPlayer.java     | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/main/java/net/minecraft/server/level/EntityPlayer.java b/src/main/java/net/minecraft/server/level/EntityPlayer.java
index 0e4af3212..089265a16 100644
--- a/src/main/java/net/minecraft/server/level/EntityPlayer.java
+++ b/src/main/java/net/minecraft/server/level/EntityPlayer.java
@@ -1033,6 +1033,18 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
         ResourceKey<DimensionManager> resourcekey = worldserver1.getTypeKey(); // CraftBukkit
 
         if (resourcekey == DimensionManager.THE_END && worldserver != null && worldserver.getTypeKey() == DimensionManager.OVERWORLD) { // CraftBukkit
+            // Origami start - add PlayerPortalEvent to leaving the end
+            Location exit = this.getBukkitEntity().getBedSpawnLocation();
+            PlayerPortalEvent event = new PlayerPortalEvent(this.getBukkitEntity(), this.getBukkitEntity().getLocation(), exit, cause, 0, true, 0);
+            Bukkit.getServer().getPluginManager().callEvent(event);
+            if (event.isCancelled()) {
+                return null;
+            }
+
+            if (event.getTo() == null || !event.getTo().equals(this.getBukkitEntity().getBedSpawnLocation())) {
+                this.getBukkitEntity().setBedSpawnLocation(event.getTo(), true);
+            }
+            // Origami end
             this.worldChangeInvuln = true; // CraftBukkit - Moved down from above
             this.decouple();
             this.getWorldServer().removePlayer(this);
-- 
2.25.1.windows.1

